Source/WebCore/Modules/reporting/IntegrityPolicyViolationReportBody.cpp

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2025 Shopify Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "IntegrityPolicyViolationReportBody.h"
+
+#include "FormData.h"
+#include <wtf/NeverDestroyed.h>
+#include <wtf/TZoneMallocInlines.h>
+
+namespace WebCore {
+
+WTF_MAKE_TZONE_OR_ISO_ALLOCATED_IMPL(IntegrityPolicyViolationReportBody);
+
+IntegrityPolicyViolationReportBody::IntegrityPolicyViolationReportBody(String&& documentURL, String&& blockedURL, String&& destination, bool reportOnly)
+    : m_documentURL(WTFMove(documentURL))
+    , m_blockedURL(WTFMove(blockedURL))
+    , m_destination(WTFMove(destination))
+    , m_reportOnly(reportOnly)
+{
+}
+
+Ref<IntegrityPolicyViolationReportBody> IntegrityPolicyViolationReportBody::create(String&& documentURL, String&& blockedURL, String&& destination, bool reportOnly)
+{
+    return adoptRef(*new IntegrityPolicyViolationReportBody(WTFMove(documentURL), WTFMove(blockedURL), WTFMove(destination), reportOnly));
+}
+
+const String& IntegrityPolicyViolationReportBody::type() const
+{
+    static NeverDestroyed<const String> integrityPolicyReportType(MAKE_STATIC_STRING_IMPL("integrity-violation"));
+    return integrityPolicyReportType;
+}
+
+const String& IntegrityPolicyViolationReportBody::documentURL() const
+{
+    return m_documentURL;
```

Comment:
These trivial getters should go inline in the header.

Reply:
Done!